### PR TITLE
Feature/map integration

### PR DESCRIPTION
**Pull request by pavicfilip**
Adds maps:
- map integration using `flutter_map`
- enabled map animations using `flutter_map_animations`
- support for different map styles (e.g. with [Mapbox](https://docs.mapbox.com/help/tutorials/create-a-custom-style/))
- in case style URLs are not set, fallbacks to the default OpenStreetMap style

Also changed:
- added environment variables using `flutter_dotenv` (also added an `.env.example` file)
- updated some styles in theme
- updated mocked trip data
- added two new pages: trip details and trip locations

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -0,0 +1,3 @@
+MAPS_TILE_LIGHT_URL=
+MAPS_TILE_DARK_URL=
+MAPS_USER_AGENT_NAME=
```
Comment (LRadocaj) [inline .env.example:3]: do this urls contain api keys, can't user agent be hardcoded?

File: lib/features/planner/trip_details/trip_map_page.dart [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+  late AnimatedMapController animatedMapController;
+
+  @override
+  void initState() {
+    super.initState();
+    animatedMapController = AnimatedMapController(vsync: this);
+  }
+
+  @override
+  void dispose() {
```
Comment (LRadocaj) [inline lib/features/planner/trip_details/trip_map_page.dart:36]: dispose at end of class

**Review by LRadocaj — CHANGES_REQUESTED** [review]
see comments, otherwise great

File: lib/features/planner/trip_details/trip_map_page.dart (hunk 2 above)
Comment (pavicfilip) [reply lib/features/planner/trip_details/trip_map_page.dart]: Moved

**Pull request merged**